doc-Community_Collaboration_Guide/topics/creating_a_new_repository.adoc

```diff
@@ -1,11 +1,153 @@
+
+* `curl`
+* `zip`
+
+[[ccg-create-repo-synch-batchcommit-dir]]
+===== Directory Structure
+
+The batch commit example described here assumes that you have a book directory, `my-sample-book`, structured as follows:
+
+----
+my-sample-book/
+|__ fetch-upstream.sh
+|__ docinfo.xml
+|__ master.adoc
+|__ topics/
+|__ upstream/
+----
+
+Where `fetch-upstream.sh` is the batch commit script (for details, see <<ccg-create-repo-synch-batchcommit-script>>), and `upstream` is a subdirectory reserved for the upstream content.
+
+[[ccg-create-repo-synch-batchcommit-workflow]]
+===== Workflow
+
+Given that you already have a batch commit script, `fetch-upstream.sh`, the workflow for synchronizing content from the upstream repository is as follows:
+
+. Pull down the latest content from the upstream repository and put the downloaded files in the `upstream/` subdirectory:
++
+----
+$ ./fetch-upstream.sh
+----
+
+. Inspect the downloaded content (for example, using `git diff`) to check for new files, restructured content, and so on.
+If necessary, edit the book file, `master.adoc`, to incorporate new content.
+
+. Add and commit the downloaded content:
++
+----
+$ git add upstream
+$ git commit -m "Synched content from upstream"
+----
+
+[[ccg-create-repo-synch-batchcommit-script]]
+===== Sample Batch Commit Script
+
+The batch commit script, `fetch-upstream.sh`, is a simple `bash` script that pulls down the latest upstream content, copying the relevant upstream files into the `upstream` subdirectory.
+The script exploits a particular feature of GitHub, whereby GitHub provides a downloadable archive file containing the working tree for any branch in the repository.
+The archive file contains a snapshot of the _latest_ content in a particular branch and thus provides an ideal basis for synchronizing content downstream.
+
+To see how his works, consider the following sample `fetch-upstream.sh` script that was developed for synchronizing the contents of the _Apache Camel Component Reference_:
+
+----
+#!/bin/sh
+
+branch=camel-2.18.x <1>
+
+echo "info: fetching upstream content from branch: $branch"
+echo
+
+# -L = follow redirects
+curl -L -o $branch.zip https://github.com/apache/camel/archive/$branch.zip <2>
+
+# -o = overwrite, -j = flatten directory structure, -q = quiet
+unzip -qjo -d upstream $branch.zip camel-$branch/components/camel-*/src/main/docs/*.adoc <3>
+unzip -qjo -d upstream/images $branch.zip camel-$branch/components/camel-*/src/main/docs/*.png
+unzip -qjo -d upstream $branch.zip camel-$branch/camel-core/src/main/docs/*-component.adoc
+
+rm $branch.zip <4>
+----
+
+<1> Set the `branch` variable to the name of the upstream branch.
+<2> Download the archive (`.zip`) file from GitHub, which contains a snapshot of the latest content from the `$branch` branch and store it locally as the file, `$branch.zip`.
```

Comment:
suggestion: which includes a snapshot ...

Reply:
Not sure what's wrong with "contains", but "includes" sounds good to me too.

Reply:
@fbolton, includes sounds better.